server-common/src/main/java/org/apache/kafka/server/metrics/KafkaMetricsGroup.java

```diff
@@ -0,0 +1,161 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.server.metrics;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+import com.yammer.metrics.core.Gauge;
+import com.yammer.metrics.core.Histogram;
+import com.yammer.metrics.core.Meter;
+import com.yammer.metrics.core.MetricName;
+import com.yammer.metrics.core.Timer;
+import org.apache.kafka.common.utils.Sanitizer;
+
+public class KafkaMetricsGroup {
+    private final Class<?> klass;
+
+    public KafkaMetricsGroup(final Class<?> klass) {
+        this.klass = klass;
+    }
+
+    /**
+     * Creates a new MetricName object for gauges, meters, etc. created for this
+     * metrics group.
+     * @param name Descriptive name of the metric.
+     * @param tags Additional attributes which mBean will have.
+     * @return Sanitized metric name object.
+     */
+    public MetricName metricName(final String name, final Map<String, String> tags) {
+        final String pkg;
+        if (klass.getPackage() == null) {
+            pkg = "";
+        } else {
+            pkg = klass.getPackage().getName();
+        }
+        final String simpleName = klass.getSimpleName().replaceAll("\\$$", "");
```

Comment:
Can we please file a JIRA for removing this? I don't think it's required if we don't pass the class from Scala objects (versus classes).

Reply:
Do you mean, this JIRA must be implemented _after_ we don't have `KafkaMetricsGroup` instantiated in Scala objects (currently we have it in a couple of places, e.g. `Partition` and `DelayedDeleteRecordsMetrics`)? Also alternatively, it can be done by overriding `metricName` in objects

Reply:
If you pass `classOf[Partition]`, then this code is not needed. It's only needed if you pass `getClass` from within a scala `object` (not `class`).